hs_core/models.py

```diff
@@ -2178,6 +2178,9 @@ class MyResourceContentType(pages.Page, hs_core.AbstractResource):
     # allow resource that contains spam_patterns to be discoverable/public
     spam_allowlisted = models.BooleanField(default=False)
 
+    quota_holder = models.ForeignKey(User, on_delete=models.CASCADE, null=True, blank=True,
```

Comment:
thank you for this Pabitra! I'm thinking maybe we use SET()
https://docs.djangoproject.com/en/3.2/ref/models/fields/#django.db.models.SET
added that, let me know what you think

Reply:
Nevermind, I realized that `SET()` [doesn't seem to take any args](https://docs.djangoproject.com/en/3.2/_modules/django/db/models/deletion/#SET) so I will use `SET_NULL` as you suggested

Reply:
What if you make this function as an instance method of the AbstractResource class?. 
https://github.com/hydroshare/hydroshare/commit/81c5e6b595c370dc99686238344f820ba83731b7#diff-3bc7182059367a2c6cf9e96428d08a0ce66f709baec5d00090abd19683582891R2112

Reply:
Yeah I have never used models.SET() before. Seems like it has quite a few restrictions. I guess we can set it to None and for now and deal with it later to update it using management command. Do we ever delete users? What about a user being inactive? If we really want to set another user as quota holder upon user delete, maybe we can use pre delete signal for user?

Reply:
ooh pre-delete signal is a good call. AFAIK we don't ever delete users, we make them inactive. I think it's okay for an inactive user to retain their quota.

I'll add a pre-delete signal

Reply:
added signal in https://github.com/hydroshare/hydroshare/pull/5469/commits/07475949a01e42bb416cf40e7dedd85a204ef3b9

Reply:
I think it might not even matter because in my testing, deleting user also deletes all of their resources.
I don't see this cascade in the code...maybe it is 
https://github.com/hydroshare/hydroshare/blob/master/hs_core/models.py#L2119-L2124